### ISSUE TITLE
Error with self-signed sertificates

### ISSUE THREAD

**Issue by ikitommi** [mieky/jestas#7]
would be even more awesomely great if it could handle self-signed certs oob:

```
Couldn't fetch build list from https://***/jenkins/api/json?pretty=true: request to https://***/jenkins/api/json?pretty=true failed, reason: self signed certificate
```

**Comment by mieky** [comment]
Agree, wonder what it would take in practice. Ideas most welcome!

**Comment by mieky** [comment]
Apparently, passing an "agent" option to node-fetch in the options with "rejectUnauthorized" set to `false`should do the trick. Need to test this.

```
if (isSSL && options.allowSelfSigned) {
    options.rejectUnauthorized = false;
    options.agent = new https.Agent(options);
}
```
Source:
https://github.com/bitinn/node-fetch/issues/19#issuecomment-243254085
https://nodejs.org/api/https.html

**Comment by mieky** [comment]
@ikitommi, just released version 1.3.0 which should now work with the `--trust` option specified!
